noop message causes panic

## Bug report

If a document is skipped via a transform function, the Op type becomes "noop" internally which causes a problem for the Elasticsearch adaptor. Originally reported [here](https://github.com/compose/transporter/pull/95#issuecomment-277715308) by @mpnovikova

config.yaml
```yaml
nodes:  
  sourcemongo:
    type: mongodb
    uri: mongodb://USER:PWD@HOST:PORT/BedrockDocs
    namespace: BedrockDocs.normaldocs
    debug: true
  destes:
    type: elasticsearch
    uri: http://USER:PWD@HOST:PORT
    namespace: ops.normaldocs
    debug: true
```
app.js
```javascript
var pipeline = Source({name:"sourcemongo"})
  .transform({filename: "transformers/normaldoc.js", namespace: "BedrockDocs.normaldocs"})
  .save({name:"destes"}) 
```
transformer.js
```javascript
module.exports = function(msg) {
  console.log("transformer: " + JSON.stringify(msg))
  if (msg.data['type'] != 'Company') { 
    return false; 
  }
  return msg;
}
```

### System info:

- Transporter version
0.1.2
- OS
any
- DB version(s)
MongoDB and Elasticsearch

### Reproducible Steps:

run transporter

### What did you expect to happened?

only a subset of documents should be sent to elasticsearch from mongodb

### What actually happened?

transporter panicked and crashed

```
transporter run --config normaldocs.yaml normaldocs.js 
INFO[0000] SYNC Processing db1:27017...                 
INFO[0000] Establishing new connection to db1:27017 (timeout=5s)... 
INFO[0000] failed to parse duration, , falling back to default timeout of 30s 
INFO[0000] Connection to db1:27017 established.         
INFO[0000] adaptor Starting...                           path=sourcemongo
INFO[0000] starting Read func                            db=BedrockDocs
INFO[0000] boot map[sourcemongo:mongodb ccbc3c47-c46f-48f5-6027-13791acdf865:transformer destes:elasticsearch]  ts=1486400048134095445
INFO[0000] adaptor Listening...                          path="sourcemongo/ccbc3c47-c46f-48f5-6027-13791acdf865/destes"
INFO[0000] Establishing new connection to db0:27017 (timeout=1h0m0s)... 
INFO[0000] SYNC Adding db1:27017 to cluster as a slave. 
INFO[0000] Connection to db0:27017 established.         
INFO[0000] SYNC Synchronization was complete (got data from primary). 
INFO[0000] SYNC Synchronization completed: 1 master(s) and 1 slave(s) alive. 
INFO[0000] collection count                              db=BedrockDocs num_collections=5
INFO[0000] skipping iteration...                         collection="customer_doc_counts" db=BedrockDocs
INFO[0000] skipping iteration...                         collection=docstats db=BedrockDocs
INFO[0000] skipping iteration...                         collection="ipec_results" db=BedrockDocs
INFO[0000] sending for iteration...                      collection=normaldocs db=BedrockDocs
INFO[0000] skipping iteration...                         collection=system.indexes db=BedrockDocs
INFO[0000] done iterating collections                    db=BedrockDocs
INFO[0000] iterating...                                  collection=normaldocs
INFO[0000] Establishing new connection to db0:27017 (timeout=1h0m0s)... 
INFO[0000] Connection to db0:27017 established.         
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xbec242]

goroutine 47 [running]:
panic(0xe36080, 0xc42000e070)
	/usr/local/Cellar/go/1.7.4/libexec/src/runtime/panic.go:500 +0x1a1
github.com/compose/transporter/vendor/gopkg.in/olivere/elastic%2ev5.(*BulkService).estimateSizeInBytes(0xc420082d10, 0x0, 0x0, 0xc4205f6be0)
	/Users/JP/gocode/src/github.com/compose/transporter/vendor/gopkg.in/olivere/elastic.v5/bulk.go:147 +0x22
github.com/compose/transporter/vendor/gopkg.in/olivere/elastic%2ev5.(*BulkService).EstimatedSizeInBytes(0xc420082d10, 0xc420023718)
	/Users/JP/gocode/src/github.com/compose/transporter/vendor/gopkg.in/olivere/elastic.v5/bulk.go:137 +0x9d
github.com/compose/transporter/vendor/gopkg.in/olivere/elastic%2ev5.(*bulkWorker).commitRequired(0xc42026db80, 0xc420023708)
	/Users/JP/gocode/src/github.com/compose/transporter/vendor/gopkg.in/olivere/elastic.v5/bulk_processor.go:539 +0x61
github.com/compose/transporter/vendor/gopkg.in/olivere/elastic%2ev5.(*bulkWorker).work(0xc42026db80)
	/Users/JP/gocode/src/github.com/compose/transporter/vendor/gopkg.in/olivere/elastic.v5/bulk_processor.go:439 +0x1d6
created by github.com/compose/transporter/vendor/gopkg.in/olivere/elastic%2ev5.(*BulkProcessor).Start
	/Users/JP/gocode/src/github.com/compose/transporter/vendor/gopkg.in/olivere/elastic.v5/bulk_processor.go:297 +0x335
```

--- a/pkg/adaptor/transformer/transformer.go
+++ b/pkg/adaptor/transformer/transformer.go
@@ -6,13 +6,13 @@ import (
 	"regexp"
 	"time"
 
+	"github.com/compose/mejson"
 	"github.com/compose/transporter/pkg/adaptor"
 	"github.com/compose/transporter/pkg/message"
 	"github.com/compose/transporter/pkg/message/adaptor/transformer"
 	"github.com/compose/transporter/pkg/message/data"
 	"github.com/compose/transporter/pkg/message/ops"
 	"github.com/compose/transporter/pkg/pipe"
-	"github.com/compose/mejson"
 	"github.com/robertkrimen/otto"
 	_ "github.com/robertkrimen/otto/underscore" // enable underscore
 )
@@ -236,7 +236,7 @@ func (t *Transformer) toMsg(origMsg message.Msg, incoming interface{}) (message.
 		}
 	case bool: // skip this doc if we're a bool and we're false
 		if !newMsg {
-			op = ops.Noop
+			return nil, nil
 		}
 	default: // something went wrong
 		return nil, fmt.Errorf("returned doc was not a map[string]interface{}: was %T", newMsg)

--- a/pkg/adaptor/transformer/transformer_test.go
+++ b/pkg/adaptor/transformer/transformer_test.go
@@ -102,7 +102,7 @@ func TestTransformOne(t *testing.T) {
 			"we should be able to skip a nil message",
 			"module.exports=function(doc) { return false }",
 			message.MustUseAdaptor("transformer").From(ops.Insert, "database.collection", data.Data{"id": bsonID1, "name": "nick"}),
-			message.MustUseAdaptor("transformer").From(ops.Noop, "database.collection", data.Data{"id": bsonID1, "name": "nick"}),
+			nil,
 			false,
 		},
 		{
@@ -138,6 +138,9 @@ func TestTransformOne(t *testing.T) {
 }
 
 func isEqual(m1 message.Msg, m2 message.Msg) bool {
+	if m1 == nil && m2 == nil {
+		return true
+	}
 	if m1.ID() != m2.ID() {
 		return false
 	}

